runtime/src/iree/hal/local/executable_environment.c

```diff
@@ -44,6 +19,15 @@ void iree_hal_executable_environment_initialize(
   IREE_ASSERT_ARGUMENT(out_environment);
   IREE_TRACE_ZONE_BEGIN(z0);
   memset(out_environment, 0, sizeof(*out_environment));
-  iree_hal_processor_query(temp_allocator, &out_environment->processor);
+
+  // Force CPU initialization.
+  // TODO(benvanik): move this someplace better? Technically not thread-safe
+  // but should be enough for usage within the HAL.
+  iree_cpu_initialize(temp_allocator);
+
+  // Will fill all of the required fields and zero any extras.
+  iree_cpu_read_data(IREE_HAL_PROCESSOR_DATA_CAPACITY_V0,
+                     &out_environment->processor.data[0]);
```

Comment:
do we really need to have the cpu data stored twice in memory - in `out_environment->processor.data` and in `iree_cpu_data_cache_` ?